Sources/AxisContribution/ViewModel/ACDataProvider.swift

```diff
@@ -35,31 +35,22 @@ public class ACDataProvider {
     ///   - constant: Settings that define the contribution view.
     ///   - sourceDates: An array of contributed dates.
     /// - Returns: mapped data
-    public func mappedData(constant: ACConstant, source sourceDates: [Date]) -> [[ACData]] {
+    public func mappedData(constant: ACConstant, source sourceDates: [Date: ACData]) -> [[ACData]] {
         var newDatas = [[ACData]]()
         var dateWeekly = Date.datesWeekly(from: constant.fromDate, to: constant.toDate)
         if constant.axisMode == .vertical {
             dateWeekly = dateWeekly.reversed()
         }
         dateWeekly.forEach { date in
             let datas = date.datesInWeek.map { date -> ACData in
-                let data = ACData(date: date, count: getDateCount(sourceDates: sourceDates, date: date))
-                return data
+                if let data = sourceDates[date] {
```

Comment:
This change brings this bug https://github.com/jasudev/AxisContribution/issues/14 in. The `date` of `sourceDates[date]` doesn't necessarily need to be the start moment of the day. Therefore, `sourceDates[date]` is nil for `Date()` when `toDate` is `Date()`

Reply:
@Maples7

Reply:
@racechao That's ok, I've created a PR #13 to fix these issues.

1. We can turn the `[Date]` into a `[Date:ACData]` internally with only 1 literation (O(n)) by doing so https://github.com/jasudev/AxisContribution/pull/13/commits/32702733644bfcc78f8798c551e3e94ecbbfc39c#diff-6eace7fd00e64a79f64bb279676278b16eab8ec0f93e83fa4d8f5a422ffd040b, I think the performance should be acceptable. The difference is whether we need to request users to make the conversion by themselves. And we'd better not to break the API policy for backward compatibility. Or, we can create a new `init` to support `[Data:ACData]` type parameter as the `sourceDatas`.
2.  Yeah the `date` keys have been turned into `startOfDay` in the 1st point.

FYI @jasudev for more context.